spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -302,14 +302,12 @@ public PlanningMode dataPlanningMode() {
       return LOCAL;
     }
 
-    String modeName =
-        confParser
-            .stringConf()
-            .sessionConf(SparkSQLProperties.DATA_PLANNING_MODE)
-            .tableProperty(TableProperties.DATA_PLANNING_MODE)
-            .defaultValue(TableProperties.PLANNING_MODE_DEFAULT)
-            .parse();
-    return PlanningMode.fromName(modeName);
+    return confParser
+        .enumConf(PlanningMode::fromName)
+        .sessionConf(SparkSQLProperties.DATA_PLANNING_MODE)
+        .tableProperty(TableProperties.DATA_PLANNING_MODE)
+        .defaultValue(PlanningMode.fromName(TableProperties.PLANNING_MODE_DEFAULT))
```

Comment:
We can also offer `defaultValue(String value)` and call `toEnum(value)` internally.